pkg/operation/operation_test.go

```diff
@@ -82,7 +82,40 @@ var _ = Describe("operation", func() {
 			"fooShoot",
 			"barProject",
 			"ingress.seed.example.com",
-			Equal("t--barProject--fooShoot.ingress.seed.example.com"),
+			Equal("t-barProject--fooShoot.ingress.seed.example.com"),
+		),
+	)
+
+	DescribeTable("#ComputeIngressHost", func(prefix, technicalID, domain string, matcher gomegatypes.GomegaMatcher) {
+		var (
+			seed = &gardencorev1beta1.Seed{
+				Spec: gardencorev1beta1.SeedSpec{
+					DNS: gardencorev1beta1.SeedDNS{
+						IngressDomain: &domain,
+					},
+				},
+			}
+			shoot = &gardencorev1beta1.Shoot{}
+			o     = &Operation{
+				Seed:  &seedpkg.Seed{},
+				Shoot: &shootpkg.Shoot{},
+			}
+		)
+
+		shoot.Status = gardencorev1beta1.ShootStatus{
+			TechnicalID: technicalID,
+		}
+
+		o.Seed.SetInfo(seed)
+		o.Shoot.SetInfo(shoot)
+
+		Expect(o.ComputeIngressHost(prefix)).To(matcher)
+	},
```

Comment:
Why do you copy this entire table instead of just adding a new `Entry` above?

Also, can you please add an entry with a technical ID starting with `shoot--`?

Reply:
Thank you for the insight! I improved the test case.
The `shoot--` case is now covered twice: with a pre-stored TechnicalID and with a freshly computed TechnicalID (without a stored TechnicalID).